### DISCUSSION TITLE
Outdated or Inaccessible VCS

### DISCUSSION THREAD

**Discussion by randygeyer** [oss-review-toolkit/ort#9502] [Q&A]
Hi ORT team,

What's the best way to deal with the types of scanner failures below due to old or inaccessible VCS systems?  I'd prefer to simply skip the scan when the Analyzer resolves these packages to outdated VCS systems like SourceForge, since it takes the scanner/downloader 30-60s to timeout on them.  Assuming there's no workaround to locate the source, is there a way to configure the scanner to skip them, or skip known outdated/bad VCS urls?  I can work around this by checking the Analyzer provenance prior to scanning, but would like to know the ***approved solution***.  

Please let me know if I'm missing something more fundamental like configuring additional VCS providers and/or curations prior to scanning.  

Finally, is there a known list of VCS urls that will fail (no longer exist), for all package managers, not just Java/Maven?

Thanks!

```[2024-11-25T06:23:53.823728800Z [ERROR]: scanner - IOException: Could not resolve provenance for package 'Maven:jfree:jcommon:1.0.15' for source code origins [ARTIFACT, VCS].
Resolution of VCS failed with:
IOException: Could not determine VCS for type 'cvs' and URL ssh://pserver:anonymous@jfreechart.cvs.sourceforge.net:/cvsroot/jfreechart/jcommon.]```

**Comment by sschuberth** [comment]
> Assuming there's no workaround to locate the source

There is. We call them "package curations" as they are able to curate metadata errors, like the declared VCS location, for packages. See the [docs](https://oss-review-toolkit.org/ort/docs/configuration/package-curations) and real-life [use-cases](https://github.com/oss-review-toolkit/ort-config/tree/main/curations).

> Finally, is there a known list of VCS urls that will fail (no longer exist), for all package managers, not just Java/Maven?

It's implied by existing VCS curations: Simply put, any VCS URL for which there is a curation would fail otherwise.

**Comment by randygeyer** [comment]
Yeah, I knew about ort-config and curations, but the ones I keep encountering don't have curations there, like the one I provided above.  Should I be using the other curation sources for better coverage?

**Reply by sschuberth** [reply]
Well, you could try with enabling other curation sources, but ideally you'd just create a new curation and contribute it to the [ort-config ones](https://github.com/oss-review-toolkit/ort-config/tree/main/curations). Maintaining curation is a community effort (same goes for [ClearlyDefined](https://docs.clearlydefined.io/docs/get-involved/data-curation) and other sources).

**Reply by sschuberth** [reply]
If you can share for which version of `jfreechart` you're trying to download the source code, I could probably quickly create a public curation for your reference.
